sygnal/database.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 matrix.org
+# Copyright 2019 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import sqlite3
+from concurrent.futures import ThreadPoolExecutor
+
+from twisted.internet.defer import Deferred
+
+logger = logging.getLogger(__name__)
+
+
+class Database:
```

Comment:
```suggestion
class Database(object):
```

It's conventional to always derive from object. (In py2 if you don't then it uses different semantics)

Reply:
> It's conventional to always derive from object

only if you want to support py2, imho